packages/back-end/src/models/SdkConnectionModel.ts

```diff
@@ -0,0 +1,360 @@
+    proxyHost: z.string().optional(),
+  })
+  .strict();
+
+export async function editSDKConnection(
+  connection: SDKConnectionInterface,
+  updates: EditSDKConnectionParams
+) {
+  const {
+    proxyEnabled,
+    proxyHost,
+    ...otherChanges
+  } = editSDKConnectionValidator.parse(updates);
+
+  const newProxy = {
+    ...connection.proxy,
+  };
+  if (proxyEnabled !== undefined && proxyEnabled !== connection.proxy.enabled) {
+    newProxy.enabled = proxyEnabled;
+  }
+  if (proxyHost !== undefined && proxyHost !== connection.proxy.host) {
+    newProxy.host = proxyHost;
+
+    const res = await testProxyConnection(
+      {
+        ...connection,
+        proxy: addEnvProxySettings(newProxy),
+      },
+      false
+    );
+    newProxy.connected = !res.error;
+    newProxy.proxyVersion = res.version;
+  }
+
+  await SDKConnectionModel.updateOne(
+    {
+      organization: connection.organization,
+      id: connection.id,
+    },
+    {
+      $set: {
+        ...otherChanges,
+        proxy: addEnvProxySettings(newProxy),
+        dateUpdated: new Date(),
+      },
+    }
+  );
+}
+
+export async function deleteSDKConnectionById(
+  organization: string,
+  id: string
+) {
+  await SDKConnectionModel.deleteOne({
+    organization,
+    id,
+  });
+}
+
+export async function markSDKConnectionUsed(key: string) {
+  await SDKConnectionModel.updateOne(
+    { key },
+    {
+      $set: {
+        connected: true,
+      },
+    }
+  );
+}
+
+export async function setProxyError(
+  connection: SDKConnectionInterface,
+  error: string
+) {
+  await SDKConnectionModel.updateOne(
+    {
+      organization: connection.organization,
+      id: connection.id,
+    },
+    {
+      $set: {
+        "proxy.error": error,
+        "proxy.connected": false,
+        "proxy.lastError": new Date(),
+      },
+    }
+  );
+}
+
+export async function testProxyConnection(
+  connection: SDKConnectionInterface,
+  updateDB: boolean = true
+): Promise<ProxyTestResult> {
+  const proxy = connection.proxy;
+  if (!proxy || !proxy.enabled || !proxy.host) {
+    return {
+      status: 0,
+      body: "",
+      error: "Proxy connection is not enabled",
+      version: "",
+      url: "",
+    };
+  }
+
+  // TODO: Is this the real endpoint we want to use?
+  const url = proxy.host.replace(/\/*$/, "") + "/healthcheck";
+  let statusCode = 0,
+    body = "";
+  try {
+    const { responseWithoutBody, stringBody } = await cancellableFetch(
+      url,
+      {
+        method: "GET",
+      },
+      {
+        maxTimeMs: 5000,
+        maxContentSize: 500,
+      }
+    );
+    statusCode = responseWithoutBody.status;
+    body = stringBody;
+
+    if (!responseWithoutBody.ok || !stringBody) {
+      return {
+        status: statusCode,
+        body: body,
+        error: "Proxy healthcheck returned a non-successful status code",
+        version: "",
+        url,
+      };
+    }
+
+    // TODO: Is this the actual response format we want to use?
```

Comment:
we also have `ok: true`, but could change this to provide more fine-grained status. Right now though, it will always be `ok: true` if we're getting a 200